proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/sdk/PowerAuthSDK.java

```diff
@@ -2560,37 +2576,111 @@ public void onDataSignedFailed(@NonNull Throwable t) {
+    }
+
+    /**
+     * Create application or activation scoped ECIES encryptor.
+     * @param context Android context, required for activation scoped encryptor.
+     * @param listener Listener with the callback methods.
+     * @param applicationScope If {@code true} then encryptor in application scope is created.
+     * @return {@link ICancelable} operation in case that the temporary encryption key needs to be acquired from the server. If the key is already
+     *         present, then returns {@code null}.
+     */
+    private @Nullable ICancelable createEciesEncryptor(@Nullable final Context context, @NonNull final IGetEciesEncryptorListener listener, final boolean applicationScope) {
+        final IPrivateCryptoHelper helper = getCryptoHelper(context);
+        return mKeystoreService.createKeyForEncryptor(applicationScope ? EciesEncryptorScope.APPLICATION : EciesEncryptorScope.ACTIVATION, helper, new ICreateKeyListener() {
+            @Override
+            public void onCreateKeySucceeded() {
+                try {
+                    final EciesEncryptor encryptor = helper.getEciesEncryptor(applicationScope ? EciesEncryptorId.GENERIC_APPLICATION_SCOPE : EciesEncryptorId.GENERIC_ACTIVATION_SCOPE);
+                    listener.onGetEciesEncryptorSuccess(encryptor);
+                } catch (PowerAuthErrorException exception) {
+                    listener.onGetEciesEncryptorFailed(exception);
+                }
+            }
+
+            @Override
+            public void onCreateKeyFailed(@NonNull Throwable throwable) {
+                listener.onGetEciesEncryptorFailed(throwable);
+            }
+        });
+    }
+
+    /**
+     * Creates a new instance of ECIES encryptor suited for application's general end-to-end encryption purposes.
+     * The returned encryptor is cryptographically bound to the PowerAuth configuration, so it can be used
      * with or without a valid activation. The encryptor also contains an associated {@link io.getlime.security.powerauth.ecies.EciesMetadata}
      * object, allowing you to properly setup HTTP header for the request.
+     * <p>
+     * Note that this method is deprecated because doesn't guarantee that encryptor is provided, or the temporary encryption
+     * key is still valid. You should use new {@link #getEciesEncryptorForApplicationScope(IGetEciesEncryptorListener)}
+     * as a replacement.
      *
      * @return New instance of {@link EciesEncryptor} object with an associated {@link io.getlime.security.powerauth.ecies.EciesMetadata}.
      * @throws PowerAuthErrorException if {@link PowerAuthConfiguration} contains an invalid configuration.
      *         You can call {@link PowerAuthErrorException#getPowerAuthErrorCode()} to get a more
      *         detailed information about the failure.
+     * @deprecated Use {@link #getEciesEncryptorForApplicationScope(IGetEciesEncryptorListener)} as a replacement.
      */
+    @Deprecated // 1.9.0
```

Comment:
We have this policy since the beginning of the SDK development. In my own experience, it's better to have a working build immediately after the upgrade and then slowly fix the warnings. This of course expects, that the developers have some common standards applied, such as zero warnings build.